core/code/hooks.js

```diff
@@ -99,11 +102,16 @@ window.removeHook = function(event, callback) {
     throw new Error('Callback must be a function.');
   }
 
-  if (_hooks[event]) {
-    var index = _hooks[event].indexOf(callback);
+  var listeners = _hooks[event];
+  if (listeners) {
+    var index = listeners.indexOf(callback);
     if(index == -1)
       log.warn('Callback wasn\'t registered for this event.');
     else
-      _hooks[event].splice(index, 1);
+      if (isRunning) {
+        listeners[index] = L.Util.falseFn;
```

Comment:
`L.Util.falseFn` always return false -> false here means 'interrupt hook'

```suggestion
        listeners[index] = function() {};
```

Reply:
Thank you for this correction!

Solution with `L.Util.falseFn;` had one advantage: it is one function used in multiple cases, in order to decrease garbage collection.

So I will change it to `listeners[index] = $.noop;`